### ISSUE TITLE
Git authentication doesn't work

### ISSUE THREAD

**Issue by Nisaba** [prominic/Moonshine-IDE#614]
I set "Need Authentication" option.

But it doesn't take the user name I set. It takes my Window login instead.
See pictures.

![image](https://user-images.githubusercontent.com/34550856/63275613-5e522900-c291-11e9-8ec2-3d0c97ccb7a4.png)

![image](https://user-images.githubusercontent.com/34550856/63275632-64e0a080-c291-11e9-8be4-f20493ade636.png)

**Comment by piotrzarzycki21** [comment]
Could you try non SSH solution ?

**Comment by Nisaba** [comment]
The git repository is hosted on a redmine server so the repository can't be reached by https

SSH access works well on Visual Studio

**Comment by piotrzarzycki21** [comment]
> The git repository is hosted on a redmine server so the repository can't be reached by https
> 
> SSH access works well on Visual Studio

Thanks. We will investigate that stuff probably next week due to vacation time.

**Comment by Nisaba** [comment]
OK, thank you very much

**Comment by rat-moonshine** [comment]
@Nisaba Thank you for reporting. Few quick questions:

1. Is "cbern" any user-name that you have previously used for 'any' Git repository?
2. Go to `Run -> Credential Manager -> Windows Credentials` - can you check if you have any Git credential saved previously, and if the user-name mentioned under the entry matches "cbern"
3. Try running `git config user.name` on command-line - what output you see?

**Comment by Nisaba** [comment]
Hi,
It takes cbern but it's my windows account that has no deal with my Git repository.
I want it to use my git user nisaba as the screenchot shows but it has no effect.
I don't want it to use my windows account.

git config user.name -> nisaba

But there are 2 git installs in my system :
- C:\Program Files\Git  (the default one I think)
- D:\MoonshineSDKs\Git\git-2.20.1

Thanks

**Comment by piotrzarzycki21** [comment]
@rat-moonshine moving discuss here from priv. You don't need windows to test that. Try access one of our repository trough ssh - did it work? You should be able to reproduce that issue. OS doesn't matter - you can use Mac.

**Comment by rat-moonshine** [comment]
> @rat-moonshine moving discuss here from priv. You don't need windows to test that. Try access one of our repository trough ssh - did it work? You should be able to reproduce that issue. OS doesn't matter - you can use Mac.

Yes, I'm investigating on mac only. Some more questions I probably ask to Nisaba again, I'll return to that shortly. OS does matter as in cases messages are different between the OS, and triggering auth window is also different between the OS'.

**Comment by rat-moonshine** [comment]
Okay. Here's a few things after I going through the current codes. 

Git requires to enter credentials in mostly two areas - 1. when cloning a private repository, 2. `git push`. In Moonshine, asking credentials happens in two ways - 1. on Windows git opens-up its native authentication window (this is automatic), 2. on macOS we provides an authentication window to enter credentials. 

@Nisaba I'm not sure during which activity you received that error - git-clone? If against git-push, what you see the user-name in the Moonshine git-commit window?

![image](https://user-images.githubusercontent.com/24370870/64686628-93bff000-d4a6-11e9-8cf0-74de7f29824c.png)

(Above fetched by the system's git-configuration. Do you see any reference of your Windows' user-name there? One more thing, what you receive if you run 'git config --global user.name'?)

Unfortunately, I didn't found we had any handling to git-clone with credentials. This probably ran automatic on Windows by popping-up native authentication window (by Git), but I didn't found any handling for macOS. I'm working on adjusting the codes now on this (I'll probably create a separate issue for the same, later). However, this may take long as requesting to `ssh` protocol with credentials are different/tricky than to `http/s`.

The user-name that you provide in Add/Edit Repository window that mainly used during SVN checkout. 

In any way, I couldn't found a possible answer why its taking your Windows' user-name @Nisaba . I'm curious to know what you're having if you ran the Git command from command-line. Remember, if a credential saved to Widnows' keychain Git tries to use it first.

**Comment by Nisaba** [comment]
Hi,
The error comes with - git-clone

as previously said, here is the result of command-line:
git config user.name
-> nisaba
